js/foundation.magellan.js

```diff
@@ -109,18 +110,13 @@ class Magellan {
    * @function
    */
   scrollToLoc(loc) {
-    // Do nothing if target does not exist to prevent errors
-    if (!$(loc).length) {return false;}
     this._inTransition = true;
-    var _this = this,
-        scrollPos = Math.round($(loc).offset().top - this.options.threshold / 2 - this.options.barOffset);
-
-    $('html, body').stop(true).animate(
-      { scrollTop: scrollPos },
-      this.options.animationDuration,
-      this.options.animationEasing,
-      function() {_this._inTransition = false; _this._updateActive()}
-    );
+    var _this = this;
+
+    Foundation.SmoothScroll.scrollToLoc(loc, this.options, function() {
```

Comment:
Based on you comment above:

> Renaming this to offset would also require renaming the barOffset default variable name of Magellan

I do not think it is a good idea to pass an array of options from components to components:
- All Magelan options are not SmoothSettings options.
- Some options could expect the same value but under a different name.
- Some options could have the same name but expect different values.
- It is generally helpful to see which options are passed when calling a component.

And the main reason behind all the previous ones: Options redirection is an exception, not the norm, and should be explicit to ensure it is correctly maintained.